mongoose.c

```diff
@@ -9308,12 +9308,12 @@ void ETH_IRQHandler(void) {
       if (++s_rxno >= ETH_DESC_CNT) s_rxno = 0;
     }
   }
-#ifdef __riscv
-  ETH->DMASR = ~0;   // TODO: do more fine-grained flag cleanup
-#else
-  ETH->DMASR = MG_BIT(7);  // Clear possible RBUS while processing
-#endif
-  ETH->DMARPDR = 0;  // and resume RX
+  // Cleanup flags
+  ETH->DMASR = (7UL < 20)    // TPS - Transmit process status
+               | (7UL < 17)  // RPS - Receive process status
+               | MG_BIT(16)  // NIS, normal interrupt summary
```

Comment:
> Bit 16 NIS: Normal interrupt summary
The normal interrupt summary bit value is the logical OR of the following when thecorresponding interrupt bits are enabled in the ETH_DMAIER register:
– ETH_DMASR [0]: Transmit interrupt
– ETH_DMASR [2]: Transmit buffer unavailable
– ETH_DMASR [6]: Receive interrupt
– ETH_DMASR [14]: Early receive interrupt
Only unmasked bits affect the normal interrupt summary bit.
This is a sticky bit and it must be cleared (by writing a 1 to this bit) each time a corresponding bit that causes NIS to be set is cleared.

This bit has been cleared above when this comes as part of the RX function.
It will be set because of TX, and that will be handled in TX, if it is set because of RX, then it will be handled on the next interrupt. The only condition that is not handled is "Early receive interrupt", which I think is not enabled, but this chip may be different.
If setting this bit solves your problem, please keep it wrapped, there is no reason to do this with the STM